### PR TITLE
CoreCLR - Remove preemptive checks for path being too long

### PR DESCRIPTION
**Pull request by steji113**
Don't preemptively validate path length as it may very well be wrong. Instead let the native interop handle it. See dotnet/corefx#8655

After this, I will be working on any CoreFX tests that need to be fixed up.

cc @JeremyKuhne

### CODE REVIEW

**Comment by steji113** [comment]
@dotnet-bot Windows_NT corefx_baseline please

**Comment by steji113** [comment]
@dotnet-bot test Windows_NT corefx_baseline please

**Comment by steji113** [comment]
@dotnet-bot help

**Comment by steji113** [comment]
@JeremyKuhne according to #9715 is CI actually running the latest CoreFX tests against my changed CoreCLR bits? I think when I initially opened this PR you hadn't merged dotnet/corefx#22131 yet though.

**Comment by steji113** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please

**Comment by steji113** [comment]
@dotnet-bot test Windows_NT x64 corefx_baseline please

**Comment by steji113** [comment]
@dotnet-bot test Windows_NT x86 corefx_baseline please

**Comment by steji113** [comment]
@JeremyKuhne It looks like I will be seeing some failures against the latest CoreFX. I believe this is to be expected until dotnet/corefx#22190 is merged in. So far it is all of the tests that I updated that are failing.

**Comment by JeremyKuhne** [comment]
> It looks like I will be seeing some failures against the latest CoreFX. I believe this is to be expected until dotnet/corefx#22190 is merged in. So far it is all of the tests that I updated that are failing.

Looks that way.

File: src/mscorlib/shared/System/IO/PathHelper.Windows.cs [hunk 1]
```diff
@@ -50,12 +50,6 @@ internal static string Normalize(string path, bool checkInvalidCharacters, bool
                 // Trim whitespace off the end of the string. Win32 normalization trims only U+0020.
                 fullPath.TrimEnd(PathInternal.s_trimEndChars);
 
-                if (fullPath.Length >= PathInternal.MaxLongPath)
```
Comment (jkotas) [inline src/mscorlib/shared/System/IO/PathHelper.Windows.cs]: Delete definitions of MaxLongPath and MaxComponentLength since they are not longer used?

Comment (steji113) [inline src/mscorlib/shared/System/IO/PathHelper.Windows.cs]: Done.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test

JIT_Generics._Coverage_chaos65204782cs_chaos65204782cs_._Coverage_chaos65204782cs_chaos65204782cs_cmd failed

**Comment by steji113** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test please

Same test failure again. Looks like it is happening sporadically.

**Pull request merged**